src/account/address.cpp

```diff
@@ -22,6 +22,14 @@ Address::Address(std::string const &bech32Address) :
         throw std::invalid_argument(ERROR_MSG_BECH32);
 }
 
+Address& Address::operator = (const Address &address)
+{
+    this->m_pk = address.m_pk;
```

Comment:
This should call the copy constructors for `string` and `vector<u8>` under the hood, right? (therefore, no data ownership issues).

Also, as above - why overriding the copy "into" / assignment operator, and why not the copy constructor instead?

Reply:
From : https://www.cplusplus.com/reference/vector/vector/operator=/:
> The copy assignment (1) copies all the elements from x into the container (with x preserving its contents).

However, you are right, address objects should be immutable, I will delete the assignment operator and override the comparison operator